From 1924-1926, the Denver Chapter of the KKK collected 29600 entries of it's member applications in two ledgers. A worker from the Rocky Mountain News then donated these ledgers to the State Historical Society (now [History Colorado](https://coloradoencyclopedia.org/article/history-colorado-colorado-historical-society)). Around 1990, History Colorado accessioned the KKK ledgers into their research center so researchers could look at the books. History Colorado then [digitzed the ledgers](https://www.historycolorado.org/kkkledgers)in the 2000s as a way to make the data more accessible to the public. The two ledgers were made into pdf files that are accesible through hyperlinks on the History Colorado (HC) website and in the dataset itself.
 
The index file contains the data of both ledgers, making it easier [to search for KKK members](https://www.historycolorado.org/sites/default/files/media/document/2021/Quick%20Tips%20for%20Searching%20KKK%20Index.pdf) across both ledgers. This is the dataset I am using to make the pandas dataframe.

In [2]:
import pandas as pd
kkk_df= pd.read_csv("kkk-ledgers-index.csv",encoding='utf-8')


pd.options.display.max_rows=50


/var/folders/9n/gp925nr113qgy0k5zrctqkcm0000gn/T/ipykernel_92105/4090270963.py:2: DtypeWarning: Columns (33) have mixed types. Specify dtype option on import or set low_memory=False.
  kkk_df= pd.read_csv("kkk-ledgers-index.csv",encoding='utf-8')


The physical KKK membership books themselves are on view at the History Colorado Center in Downtown Denver. Recently, History Colorado created a data visualization in the form of a map for the addresses within the KKK ledgers. It maps addresses where KKK members lived, worked, and even recreated, on a map of Denver. The map is a Google Map of Denver today, so addresses are not entirely accurate because streets and buildings have moved.




<iframe src="https://www.google.com/maps/d/embed?mid=1Pyfnddh9OB0Ip5KKltGkgPXnFvDlve5s&ehbc=2E312F" width="640" height="480"></iframe>

The ledgers and dataset recorded member's names and both their work and residence address. The data is a way for people to see the prevalance of the KKK in Denver during the 1920s. Many addresses, recreation areas, and business are tied with the KKK by being listed in the ledger, such as National Banks, Fire Departments, and the State Capitol. Former Denver Mayor Ben F. Stapleton is on the list as 'BF Stapleton.' 


In [6]:
kkk_df['Notes & Remarks'] = kkk_df["Notes & Remarks"].str.lower()
kkk_df[kkk_df["fullName"].str.contains('Stapleton')]



itemID   Number                fullName Prefix  Last Name First Name  \
628    K00629   1128.0            BF Stapleton    NaN  Stapleton         BF   
14718  K14719  25717.0      Arthur W Stapleton    NaN  Stapleton     Arthur   
15504  K15505  46318.0  John Stapleton Rickman    NaN    Rickman       John   

      Middle Name Suffix   residenceAddress1 residenceAddress2  ...  \
628           NaN    NaN        430 Williams               NaN  ...   
14718           W    NaN                 NaN               NaN  ...   
15504   Stapleton    NaN  1014 East 10th Ave               NaN  ...   

          ledger              link Ledger Link Page           pdfFileName  \
628    MSS.366.5  h-co.org/ledger2   MSS.366.5   32    mss366-5_FC-p74_QC   
14718  MSS.366.5  h-co.org/ledger2   MSS.366.5  651  mss366-5_p602-676_QC   
15504  MSS.366.5  h-co.org/ledger2   MSS.366.5  687  mss366-5_p677-742_QC   

      symbolExists sNumber sErased Notes & Remarks Column 29  
628          False    True   False             NaN       NaN  
14718        False    True   False             NaN       NaN  
15504         True   False   False             NaN       NaN  

[3 rows x 34 columns]

Bringing up the Stapleton name is an example of important political and cultural context to keep in mind about this data. Namely, about the prevalance of KKK members in Colorado's 1920 politics. The KKK frequently had periods of decline and resurgence, and the 1920s is one of them. Colorado at one point had the [second highest](https://youtu.be/PO5PMbtF1t8?si=2J_gIOA9oI9aGaeM) per-capita KKK membership in United States. The surge in KKK membership came from increased xenophobia after increased immigration of Italians, Jewish people, and Mexicans to the U.S in the early 1900s. According to RMPBS [*Colorado Experience*](https://youtu.be/PO5PMbtF1t8?si=2J_gIOA9oI9aGaeM), lynchings of people of color in Colorado were common during the 1920s, including but not limited to Black, Mexican, and Italian people.

## About the Data

There are about 34 columns within the dataset. Many of the columns are extra columns for string data like business or residence addresses. Ledger links and references to what page of the ledger take up 5 columns. There are 3 columns with boolean values. The first simply seems to tell whether or not a symbol was written next to someone's name in the ledger. The last two, 'sNumber and sErased' are a bit unclear. In the physical ledgers some members have an 's' written next to their membership number. If the s is faded or erased then the index lists both the columns as true. There is no clear reason for why this is noted.

In [9]:
kkk_df[kkk_df['sErased']==True][:2] ##in place this will be a screenshot of page 28 ledgers

itemID  Number      fullName Prefix Last Name First Name Middle Name  \
530  K00531  1030.0  W W Phillips    NaN  Phillips          W           W   
539  K00540  1039.0  D K Proffitt    NaN  Proffitt          D           K   

    Suffix residenceAddress1 residenceAddress2  ...     ledger  \
530    NaN     2955 York St.               NaN  ...  MSS.366.5   
539    NaN     1617 Fillmore               NaN  ...  MSS.366.5   

                 link Ledger Link Page         pdfFileName symbolExists  \
530  h-co.org/ledger2   MSS.366.5   28  mss366-5_FC-p74_QC         True   
539  h-co.org/ledger2   MSS.366.5   28  mss366-5_FC-p74_QC         True   

    sNumber sErased Notes & Remarks Column 29  
530    True    True             NaN       NaN  
539    True    True             NaN       NaN  

[2 rows x 34 columns]

<div align="center">
  <img
    src="../assets/img/page-28-1.png"
    width="60%"
    style="border: 2px solid"
    alt="Description of image"
  />
  <figcaption style="font-style: italic">
    What Phillips's entry looks like in the ledger. Symbol written next to his name, along with an 's' next to his number that's faded out
  </figcaption>
</div>
<div align="center">
  <img
    src="../assets/img/page-28-2.png"
    width="60%"
    style="border: 2px solid"
    alt="Description of image"
  />
  <figcaption style="font-style: italic">
    Same page of the ledger, but with Proffitt's entry and another entry in comparison. Proffitt's entry has the 's' faded out.
  </figcaption>
</div>


The last two columns lists notes and annotations physically written on the ledgers. They list whatever was written on top of someone's name, such as whether they resigned, died, or banished. Column 29 is also made for this purpose, but it only has one entry of someone who resigned. This more than likely was added by History Colorado, as the ledger entry itself looks like a regular entry. 

In [12]:
kkk_df[kkk_df['Column 29'].notna()]

itemID   Number         fullName Prefix Last Name First Name  \
2204  K02205  13203.0  Arvid A Seurbom    NaN   Seurbom      Arvid   

     Middle Name Suffix residenceAddress1 residenceAddress2  ...     ledger  \
2204           A    NaN        3455 Adams               NaN  ...  MSS.366.5   

                  link Ledger Link Page          pdfFileName symbolExists  \
2204  h-co.org/ledger2   MSS.366.5  101  mss366-5_p75-149_QC         True   

     sNumber sErased Notes & Remarks          Column 29  
2204   False   False  name struck; "  RESIGNED 5-27-26"  

[1 rows x 34 columns]

A closer look at the Notes and Remarks columns can show how many people in the Greater Denver KKK had their names struck and for what reason. Reasons included banishment, death, resignation, suspension, etc. 



In [14]:
notes_df= kkk_df[kkk_df['Notes & Remarks'].notna()]

notes_df[notes_df['Notes & Remarks'].str.contains('resigned')]

itemID   Number            fullName Prefix Last Name First Name  \
69     K00070    569.0  Herbert R Borchard    NaN  Borchard    Herbert   
113    K00114    613.0    David T Chalmers    NaN  Chalmers      David   
120    K00121    620.0           J W Clark    NaN     Clark          J   
311    K00312    811.0      Frank II Heart    NaN     Heart      Frank   
316    K00317    816.0             R Hosea    NaN     Hosea          R   
...       ...      ...                 ...    ...       ...        ...   
15466  K15467  46280.0     Arthur B Porter    NaN    Porter     Arthur   
15762  K15763  46576.0          A C Pierce    NaN    Pierce          A   
15818  K15819  46632.0    Homer C Becktell    NaN  Becktell      Homer   
16071  K16072  47137.0        Ray W Colyer    NaN    Colyer        Ray   
16496  K16497  47562.0         Oscar Odell    NaN     Odell      Oscar   

      Middle Name Suffix residenceAddress1      residenceAddress2  ...  \
69              R    NaN               NaN  12101 E. Colfax Ave.*  ...   
113             T    NaN      854 Clarkson                    NaN  ...   
120             W    NaN   3125 W 21st Ave                    NaN  ...   
311            II    NaN               NaN       17th & Broadway*  ...   
316           NaN    NaN               NaN                    NaN  ...   
...           ...    ...               ...                    ...  ...   
15466           B    NaN    3436 W 8th Ave                    NaN  ...   
15762           C    NaN      1049 W Evans                    NaN  ...   
15818           C    NaN   2308 W 30th Ave                    NaN  ...   
16071           W    NaN       4100 Bryant                    NaN  ...   
16496         NaN    NaN               NaN             1011 19th*  ...   

          ledger              link Ledger Link Page           pdfFileName  \
69     MSS.366.5  h-co.org/ledger2   MSS.366.5    8    mss366-5_FC-p74_QC   
113    MSS.366.5  h-co.org/ledger2   MSS.366.5    9    mss366-5_FC-p74_QC   
120    MSS.366.5  h-co.org/ledger2   MSS.366.5   10    mss366-5_FC-p74_QC   
311    MSS.366.5  h-co.org/ledger2   MSS.366.5   18    mss366-5_FC-p74_QC   
316    MSS.366.5  h-co.org/ledger2   MSS.366.5   18    mss366-5_FC-p74_QC   
...          ...               ...         ...  ...                   ...   
15466  MSS.366.5  h-co.org/ledger2   MSS.366.5  685  mss366-5_p677-742_QC   
15762  MSS.366.5  h-co.org/ledger2   MSS.366.5  698  mss366-5_p677-742_QC   
15818  MSS.366.5  h-co.org/ledger2   MSS.366.5  700  mss366-5_p677-742_QC   
16071  MSS.366.5  h-co.org/ledger2   MSS.366.5  712  mss366-5_p677-742_QC   
16496  MSS.366.5  h-co.org/ledger2   MSS.366.5  731  mss366-5_p677-742_QC   

      symbolExists sNumber sErased  \
69           False   False   False   
113          False   False   False   
120          False    True   False   
311          False   False   False   
316          False   False   False   
...            ...     ...     ...   
15466         True   False   False   
15762         True   False   False   
15818        False   False   False   
16071         True   False   False   
16496        False   False   False   

                                    Notes & Remarks Column 29  
69                          name struck; "resigned"       NaN  
113                  name struck; "resigned 6-1-23"       NaN  
120                name struck; "resigned - 2-7-27"       NaN  
311                                     "resigned "       NaN  
316                                      "resigned"       NaN  
...                                             ...       ...  
15466                name struck; "resigned 9-1-25"       NaN  
15762  name struck; "m. m. demits resigned 6/30/26"       NaN  
15818               name struck; "resigned 1-13-26"       NaN  
16071             name struck; "resigned - 5-10-27"       NaN  
16496               name struck; "resigned 1-26-26"       NaN  

[215 rows x 34 columns]

Here, in total we have about 215 resignations of KKK members.

In [16]:
notes_df[notes_df['Notes & Remarks'].str.contains('banish')]

itemID   Number             fullName Prefix   Last Name First Name  \
550    K00551   1050.0        Robert F Reed    NaN        Reed     Robert   
816    K00817   1316.0       Daniel J Sparr    NaN       Sparr     Daniel   
1001   K01002   2000.0     Morton MacCormac    NaN   MacCormac     Morton   
1161   K01162   2160.0           G L Bogard    NaN      Bogard          G   
2828   K02829  13827.0       R E Strickland    NaN  Strickland          R   
4939   K04940  15938.0    Chas Root Cavalli    NaN     Cavalli       Chas   
6865   K06866  17864.0    Keith Scott Boehm    NaN       Boehm      Keith   
10173  K10174  21172.0         John W Perry    NaN       Perry       John   
11994  K11995  22993.0  Clarence R Anderson    NaN    Anderson   Clarence   

      Middle Name Suffix  residenceAddress1 residenceAddress2  ...     ledger  \
550             F    NaN    1329 So Lincoln               NaN  ...  MSS.366.5   
816             J    NaN         129 Stuart               NaN  ...  MSS.366.5   
1001          NaN    NaN                NaN               NaN  ...  MSS.366.5   
1161            L    NaN                NaN               NaN  ...  MSS.366.5   
2828            E    NaN                NaN               NaN  ...  MSS.366.5   
4939         Root    NaN  3985 West 1st Ave               NaN  ...  MSS.366.5   
6865        Scott    NaN       1532 Glenarm               NaN  ...  MSS.366.5   
10173           W    NaN        2290 Cherry               NaN  ...  MSS.366.5   
11994           R    NaN        751 So High               NaN  ...  MSS.366.5   

                   link Ledger Link Page           pdfFileName symbolExists  \
550    h-co.org/ledger2   MSS.366.5   29    mss366-5_FC-p74_QC        False   
816    h-co.org/ledger2   MSS.366.5   40    mss366-5_FC-p74_QC        False   
1001   h-co.org/ledger2   MSS.366.5   49    mss366-5_FC-p74_QC        False   
1161   h-co.org/ledger2   MSS.366.5   56    mss366-5_FC-p74_QC        False   
2828   h-co.org/ledger2   MSS.366.5  128   mss366-5_p75-149_QC        False   
4939   h-co.org/ledger2   MSS.366.5  221  mss366-5_p150-224_QC        False   
6865   h-co.org/ledger2   MSS.366.5  305  mss366-5_p300-374_QC        False   
10173  h-co.org/ledger2   MSS.366.5  449  mss366-5_p375-449_QC        False   
11994  h-co.org/ledger2   MSS.366.5  532  mss366-5_p526-601_QC        False   

      sNumber sErased                          Notes & Remarks Column 29  
550     False   False                 name struck; " banished"       NaN  
816     False   False                 name struck; " banished"       NaN  
1001    False   False                 name struck; " banished"       NaN  
1161    False   False        name struck; "banished- 12/29/24"       NaN  
2828     True   False                  name struck; "banished"       NaN  
4939    False   False   name struck; "banished by gd 11-25-24"       NaN  
6865    False   False     name struck; "banished by gd 1-6-25"       NaN  
10173   False   False   name struck; "banished by gd - 4/1725"       NaN  
11994   False   False  name struck; "banished by gd - 4/28/25"       NaN  

[9 rows x 34 columns]

Between 1924-26 9 people were banished from the Klan.

In [18]:
notes_df[notes_df['Notes & Remarks'].str.contains('suspend')]

itemID   Number              fullName Prefix  Last Name First Name  \
366    K00367    866.0          Homer R Koen    NaN       Koen      Homer   
649    K00650   1149.0       Roger W Taggart    NaN    Taggart      Roger   
1302   K01303   2301.0             W A Cloud    NaN      Cloud          W   
2179   K02180  13178.0             G C Minor    NaN      Minor          G   
3116   K03117  14115.0      Carl Law Mahoney    NaN    Mahoney       Carl   
3860   K03861  14859.0          Ernest Zesch    NaN      Zesch     Ernest   
4281   K04282  15280.0            V R Stokes    NaN     Stokes          V   
5984   K05985  16983.0       Curtis C Godwin    NaN     Godwin     Curtis   
6669   K06670  17668.0    Arthur B Partridge    NaN  Partridge     Arthur   
7744   K07745  18743.0   Herbert A Donaldson    NaN  Donaldson    Herbert   
9997   K09998  20996.0      Claude Edw Mason    NaN      Mason     Claude   
10389  K10390  21388.0   Clyde Lorence Magee    NaN      Magee      Clyde   
10409  K10410  21408.0            Lee R West    NaN       West        Lee   
11199  K11200  22198.0         Harold A Hays    NaN       Hays     Harold   
13376  K13377  24375.0      Jas Henry Spidle    NaN     Spidle        Jas   
13924  K13925  24923.0  Willard Ray Reynolds    NaN   Reynolds    Willard   
14446  K14447  25445.0            Earl Moore    NaN      Moore       Earl   
14850  K14851  25849.0       V Alley Brissey    NaN    Brissey          V   

      Middle Name Suffix residenceAddress1 residenceAddress2  ...     ledger  \
366             R    NaN               NaN               NaN  ...  MSS.366.5   
649             W    NaN        1356 Pearl               NaN  ...  MSS.366.5   
1302            A    NaN        1520 Ogden               NaN  ...  MSS.366.5   
2179            C    NaN               NaN  1441 Glenarm Pl*  ...  MSS.366.5   
3116          Law    NaN               NaN               NaN  ...  MSS.366.5   
3860          NaN    NaN      1360 W Byers               NaN  ...  MSS.366.5   
4281            R    NaN   1616 So Downing               NaN  ...  MSS.366.5   
5984            C    NaN   2651 W 34th Ave               NaN  ...  MSS.366.5   
6669            B    NaN          1037 Fox               NaN  ...  MSS.366.5   
7744            A    NaN       1268 Marion               NaN  ...  MSS.366.5   
9997          Edw    NaN               NaN               NaN  ...  MSS.366.5   
10389     Lorence    NaN               NaN               NaN  ...  MSS.366.5   
10409           R    NaN      1346 Sherman               NaN  ...  MSS.366.5   
11199           A    NaN   1207 California               NaN  ...  MSS.366.5   
13376       Henry    NaN       681 Vallejo               NaN  ...  MSS.366.5   
13924         Ray    NaN    4790 W 5th Ave               NaN  ...  MSS.366.5   
14446         NaN    NaN  1933 Arapahoe St               NaN  ...  MSS.366.5   
14850       Alley    NaN  1326 Cherokee St               NaN  ...  MSS.366.5   

                   link Ledger Link Page           pdfFileName symbolExists  \
366    h-co.org/ledger2   MSS.366.5   21    mss366-5_FC-p74_QC        False   
649    h-co.org/ledger2   MSS.366.5   33    mss366-5_FC-p74_QC        False   
1302   h-co.org/ledger2   MSS.366.5   62    mss366-5_FC-p74_QC        False   
2179   h-co.org/ledger2   MSS.366.5  100   mss366-5_p75-149_QC         True   
3116   h-co.org/ledger2   MSS.366.5  142   mss366-5_p75-149_QC        False   
3860   h-co.org/ledger2   MSS.366.5  174  mss366-5_p150-224_QC        False   
4281   h-co.org/ledger2   MSS.366.5  193  mss366-5_p150-224_QC        False   
5984   h-co.org/ledger2   MSS.366.5  267  mss366-5_p225-299_QC        False   
6669   h-co.org/ledger2   MSS.366.5  297  mss366-5_p225-299_QC        False   
7744   h-co.org/ledger2   MSS.366.5  343  mss366-5_p300-374_QC        False   
9997   h-co.org/ledger2   MSS.366.5  441  mss366-5_p375-449_QC        False   
10389  h-co.org/ledger2   MSS.366.5  459  mss366-5_p450-524_QC        Fals

Finally, 18 people were suspended from the KKK according to the ledger. Other reasons included members kicked out of the KKK because they were not naturalized U.S citizens. 

In [20]:
notes_df[notes_df['Notes & Remarks'].str.contains('naturalized')][:1]

itemID  Number   fullName Prefix Last Name First Name Middle Name Suffix  \
697  K00698  1197.0  C H Wells    NaN     Wells          C           H    NaN   

    residenceAddress1 residenceAddress2  ...     ledger              link  \
697        1643 Stout               NaN  ...  MSS.366.5  h-co.org/ledger2   

    Ledger Link Page         pdfFileName symbolExists sNumber sErased  \
697   MSS.366.5   35  mss366-5_FC-p74_QC        False   False   False   

                       Notes & Remarks Column 29  
697  name struck; " never-naturalized"       NaN  

[1 rows x 34 columns]

<div align="center">
  <img
    src="../assets/img/wells-sc-kkk.png"
    width="60%"
    style="border: 2px solid"
    alt="Description of image"
  />
  <figcaption style="font-style: italic">
    What C.H Well's entry looks like in the ledger, for reference.
  </figcaption>
</div>


Overall, the notes & remarks column are a great way for the index to show how some data was situated in the ledger itself. Without the inclusion of this column, the kkk-dataset would look more like "raw" data than actual "cooked" data because it would not show that this ledger was physically annotated by the original authors. 

Some limitations of this data include the lack of context for addresses and Denver itself. Many of the addresses and buildings referenced in the ledgers might not exist today or have moved. For example, some locations are linked to Auraria and Empower Field as both were previously neighborhoods. The area around the Bronco's Stadium in particular was a thriving Jewish neighborhood in early 20th Century. What might be missing for this data is a map of 1920s Denver that can line up with the data.